homeassistant/components/netatmo/sensor.py

```diff
@@ -839,15 +839,16 @@ def async_update_callback(self) -> None:
         elif self.entity_description.key == "guststrength":
             data = self._data.get_latest_gust_strengths()
 
-        if data is None:
-            if self.state is None:
-                return
-            _LOGGER.debug(
-                "No station provides %s data in the area %s",
-                self.entity_description.key,
-                self._area_name,
-            )
-            self._attr_native_value = None
+        if data is None or data == {}:
```

Comment:
Wouldn't `not` work here?
```suggestion
        if not data:
```

I think this change should also be implemented in
https://github.com/home-assistant/core/blob/fb3960c0a5e37d1464b8a7813f0253078cc4a24b/homeassistant/components/netatmo/sensor.py#L535